code/modules/antagonists/heretic/structures/mawed_crucible.dm

```diff
@@ -28,6 +28,8 @@
 	if(COOLDOWN_TIMELEFT(src, refill_cooldown))
 		return
 	COOLDOWN_START(src, refill_cooldown, 30 SECONDS)
+	if(current_mass >= max_mass)
```

Comment:
Move this before we start the cooldown
https://github.com/tgstation/tgstation/blob/5d0553ab090c4e0e5497fad142c80207012df84c/code/modules/antagonists/heretic/structures/mawed_crucible.dm#L28
So you know we don't waste time processing